### PR TITLE
[ZEPPELIN-6120] Jackson dependency governance

### PR DESCRIPTION
**Pull request by pan3793**
### What is this PR for?

Currently, Jackson deps versions are inconsistent, which may cause potential issues.

```
$ ls lib | grep jackson
jackson-annotations-2.12.7.jar
jackson-core-2.12.7.jar
jackson-core-asl-1.9.13.jar
jackson-databind-2.12.7.1.jar
jackson-mapper-asl-1.9.13.jar
jackson-module-jaxb-annotations-2.9.9.jar
jersey-media-json-jackson-2.30.jar
```

### What type of PR is it?

Improvement

### Todos
* [x] - Pin Jackson 2.12.7.2
* [x] - Manage all Jackson deps via BOM
* [x] - Remove redundant exclusions
* [x] - Pin `jakarta.activation:jakarta.activation-api:1.2.2` to satisfy enforce-dependency-convergence
* [x] - Pin `jakarta.xml.bind:jakarta.xml.bind-api:2.3.3` to satisfy enforce-dependency-convergence

### What is the Jira issue?

ZEPPELIN-6120

### How should this be tested?

```
$ ls lib | grep jackson
jackson-annotations-2.12.7.jar
jackson-core-2.12.7.jar
jackson-core-asl-1.9.13.jar
jackson-databind-2.12.7.2.jar
jackson-mapper-asl-1.9.13.jar
jackson-module-jaxb-annotations-2.12.7.jar
jersey-media-json-jackson-2.30.jar
```

### Questions:
* Does the license files need to update? Updated.
* Is there breaking changes for older versions? No.
* Does this needs documentation? No.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -122,7 +122,10 @@
     <gson.version>2.8.9</gson.version>
     <gson-extras.version>0.2.2</gson-extras.version>
     <org-json.version>20240205</org-json.version>
+    <jackson.version>2.12.7.20240502</jackson.version>
```
Comment (pan3793) [inline pom.xml:125]: it's the jackson-bom version https://mvnrepository.com/artifact/com.fasterxml.jackson/jackson-bom

File: pom.xml [hunk 2]
```diff
@@ -329,6 +332,26 @@
...
 
+      <dependency>
+        <groupId>com.fasterxml.jackson</groupId>
+        <artifactId>jackson-bom</artifactId>
+        <version>${jackson.version}</version>
+        <type>pom</type>
+        <scope>import</scope>
+      </dependency>
+
+      <dependency>
```
Comment (jongyoul) [inline pom.xml:343]: I have a question. What is this one?

File: pom.xml [hunk 3]
```diff
@@ -329,6 +332,26 @@
         <version>${gson-extras.version}</version>
       </dependency>
 
+      <dependency>
+        <groupId>com.fasterxml.jackson</groupId>
+        <artifactId>jackson-bom</artifactId>
```
Comment (jongyoul) [inline pom.xml:337]: I like it. 👍

File: pom.xml (hunk 2 above)
Comment (pan3793) [reply pom.xml:343]: building fails without this version pinning
```
[INFO] --- maven-enforcer-plugin:3.0.0-M3:enforce (enforce-dependency-convergence) @ zeppelin-server ---
[WARNING]
Dependency convergence error for jakarta.activation:jakarta.activation-api:1.2.1 paths to dependency are:
+-org.apache.zeppelin:zeppelin-server:0.12.0-SNAPSHOT
  +-org.glassfish.jersey.media:jersey-media-json-jackson:2.30
    +-com.fasterxml.jackson.module:jackson-module-jaxb-annotations:2.12.7
      +-jakarta.activation:jakarta.activation-api:1.2.1
and
+-org.apache.zeppelin:zeppelin-server:0.12.0-SNAPSHOT
  +-org.glassfish.jersey.core:jersey-server:2.30
    +-jakarta.xml.bind:jakarta.xml.bind-api:2.3.3
      +-jakarta.activation:jakarta.activation-api:1.2.2
```

Comment (jongyoul) [reply pom.xml:343]: Aha ~ Thank you for confirming it.

**Review by jongyoul — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merged to master/0.12

**Pull request merged**